packages/syft/src/syft/service/code/user_code.py

```diff
@@ -847,24 +869,177 @@ class UserCodeExecutionResult(SyftObject):
+                        for x in user_service.stash.partition.data.values()
+                        if x.verify_key == context.credentials
+                    ][0]
+                    user_api = node.get_api(context.credentials)
+                    user_api.signing_key = user_signing_key
+                    # We hardcode a python connection here since we have access to the node
+                    # TODO: this is not secure
+                    user_api.connection = PythonConnection(node=node)
+
+                    APIRegistry.set_api_for(
+                        node_uid=node.id,
+                        user_verify_key=context.credentials,
+                        api=user_api,
+                    )
+                    # original_print("ID", context.job.log_id)
+
+                    return job
+                except Exception as e:
+                    print(f"ERROR {e}")
+                    raise ValueError(f"error while launching job:\n{e}")
+
+        # original_print(f"EXECUTING BYTE CODE")
+
+        if context.job is not None:
+
+            def print(*args, sep=" ", end="\n"):
+                def to_str(arg: Any) -> str:
+                    if isinstance(arg, bytes):
+                        return arg.decode("utf-8")
+                    if isinstance(arg, Job):
+                        return f"JOB: {arg.id}"
+                    if isinstance(arg, SyftError):
+                        return f"JOB: {arg.message}"
+                    if isinstance(arg, ActionObject):
+                        return str(arg.syft_action_data)
+                    return arg
+
+                new_args = [to_str(arg) for arg in args]
+                new_str = sep.join(new_args) + end
+                # original_print(
+                #     f"appending to {context.job.log_id}, {id(context)}", *args
+                # )
+                log_service = context.node.get_service("LogService")
+                log_service.append(
+                    context=context, uid=context.job.log_id, new_str=new_str
+                )
+                return __builtin__.print(
+                    f"FUNCTION LOG ({context.job.log_id}):",
+                    *new_args,
+                    end=end,
+                    sep=sep,
+                    file=sys.stderr,
+                )
+
+        else:
+            print = original_print
+
+        #     return __builtin__.print(*args, **kwargs)
+
+        if code_item.uses_domain:
+            kwargs["domain"] = LocalDomainClient()
+
         stdout = StringIO()
         stderr = StringIO()
 
-        sys.stdout = stdout
-        sys.stderr = stderr
+        # sys.stdout = stdout
+        # sys.stderr = stderr
 
         # statisfy lint checker
         result = None
 
         exec(code_item.byte_code)  # nosec
+        _locals = locals()
+
+        user_code_service = context.node.get_service("usercodeservice")
+        for user_code in user_code_service.stash.get_all(context.credentials).ok():
```

Comment:
this is necessary because in the outer syft function function we pass in the the batch function name as the first argument to launch_job, e.g.
`domain.launch_job(run_batch, .....)` so this function needs to be present in the scope